### DISCUSSION TITLE
[monorepos] How to get hoisted packages installed with EAS build

### DISCUSSION THREAD

**Discussion by JordanDDisch** [expo/expo#18749] [Archived: Questions]
For users with a monorepo and using hoisted packages, how do you get them to be installed properly when running EAS build?

The problem is that when running EAS build it seems to only upload the contents of your apps/mobile folder. If we have shared dependencies like UI libraries etc... these do not get uploaded. I've managed to resolve this issues by manually installing the dependency in the apps/mobile folder so that is references in the packages.json, but optimally those hoisted references should be available without having to install them to the apps/mobile folder.

**Comment by keith-kurak** [comment]
There's an example here that demonstrates [EAS Build with a monorepo that includes a shared internal library](https://github.com/byCedric/eas-monorepo-example) that may be helpful to look at. In this example, the UI library is not in the app's **node_modules** folder, though I see it is listed in **package.json**, so not entirely sure if it's the same thing you're looking for. This example does describe a scenario where the internal library requires a separate build process, and how to do that with EAS. In short, EAS should see the app library files, but it only sees what is checked in, so hence the use of the `eas-build-post-install` command.

**Comment by senghuotlay** [comment]
this only gets fixed with pnpm, but how can we can it fixed within yarn?
